### DISCUSSION TITLE
Determine if app is installed for a batch file

### DISCUSSION THREAD

**Discussion by wally-crunk** [ScoopInstaller/Scoop#5860] [Help and Questions]
As with [this discussion](https://github.com/ScoopInstaller/Scoop/discussions/5760),  I'd like to check if an Scoop is installed --- but also check if a certain app is installed.

My goal a batch file that will
- use the scoop installation of the app if possible
- search Program Files and Program Files (x86), as the backup.

It looks like I can do
```
where scoop 
if %errorlevel% EQU 0 (
  for /f %%THE_DIRECTORY in ('scoop prefix sox') do set=%%THE_DIRECTORY )
```
But I don't seem to be able to get an ERRORLEVEL when "scoop which sox" finds **_nothing_**:
```
C:\Users\me>scoop prefix sox & echo %errorlevel%
C:\Users\me\scoop\apps\sox\current
0
C:\Users\me>scoop prefix foobar & echo %errorlevel%
Could not find app path for 'foobar'.
0
```

I don't mind if I use 'scoop prefix xxx' or 'scoop which xxx' — but neither one outputs an errorlevel that I can see.

For batch file purposes, is there a better way to check if a scoop app exists or doesn't exist?

Should I just check if my variable %%THE_DIRECTORY is a valid directory?

**Comment by niheaven** [comment]
You could use `$LASTEXITCODE` to check if there's no such app in PowerShell, and I don't know how to check this status in CMD.

**Comment by wally-crunk** [comment]
This is where I've arrived.  It would be nicer and cleaner if there was an %ERRORLEVEL% set.

I think this will work with any scoop app.
```
WHERE scoop 1>NUL
IF %errorlevel% GEQ 1 (
    ECHO Scoop is not installed...
) ELSE (
    set THE_OUTPUT=""
    FOR /f %%i in ('scoop prefix sox') do set THE_OUTPUT=%%i
        if exist !THE_OUTPUT! (
        set soxdir=!THE_OUTPUT!
        ECHO Sox found using Scoop at !soxdir!
        GOTO ready
    ) 
)
```
